src/borg/testsuite/archiver.py

```diff
@@ -408,8 +408,9 @@ def _set_repository_id(self, path, id):
             return repository.id
 
     def test_sparse_file(self):
-        # no sparse file support on Mac OS X
-        sparse_support = sys.platform != 'darwin'
+        # Mac OS X has no sparse file support
+        # Solaris (ZFS) has sparse file support, but is less predictable about it
+        sparse_support = sys.platform not in ['darwin', 'sunos5']
```

Comment:
Yes, it's technically not only about the OS but also about the FS (and possibly configuration options of it).